portal-impl/src/com/liferay/portal/service/impl/GroupLocalServiceImpl.java

```diff
@@ -861,18 +861,16 @@ public Group deleteGroup(Group group) throws PortalException {
 					group.getGroupId(), RoleConstants.TYPE_SITE);
 			}
 			else {
-				if (!group.isStagingGroup() || group.isStagedRemotely()) {
```

Comment:
> do we not want a staging site to have an Owner?
Staging site is only an envelope for staging data. It should have no membership assignments, no roles assignments. Even SITE_OWNER is inherited from live site.

But as I said, it should be fine to remove it, there should be no assignments anyway.